pkgs/top-level/all-packages.nix

```diff
@@ -723,6 +723,8 @@ in
 
   git-fire = callPackage ../tools/misc/git-fire { };
 
+  git-ignore = callPackage ../tools/misc/git-ignore { };
```

Comment:
A bit unsure about where I should place it, it might be a better fit in `development` over `tools`, but I am not sure about the separation.

Reply:
I would suggest to put it under `applications/version-management/git-and-tools` and include it in `applications/version-management/git-and-tools/default.nix` instead of `all-packages.nix`. That's where most git-related tools are.

Reply:
This means that to install it you have to do `nix-env -iA nixos.gitAndTools.git-ignore`, should I add it to `pkgs/top-level/aliases.nix` file too in that case?

Reply:
There seems to be no hard rule for that. Currently, the majority of git-related packages is only accessible via `gitAndTools` and installed like you mentioned. However,

- some are inherited on the top-level in `all-packages.nix`
- one is aliased in `aliases.nix` - which is wrong IMO as we use aliases mostly for older, deprecated names, not as convenience shortcuts

My recommendation is to avoid duplication and only put your package in `gitAndTools`.

Reply:
Alright, done! 😃